services/ml/compilation_impl.cc

```diff
@@ -209,10 +211,51 @@ int32_t CompilationImpl::GetConvParams(const mojom::OperationPtr& operation,
   }
   params.filter_height = filter->dimensions[1];
   params.filter_width = filter->dimensions[2];
+  params.filter_type = filter->type;
 
   const uint32_t bias_idx = inputs[index++];
   mojom::OperandPtr& bias = model_info_->operands[bias_idx];
   params.bias_length = bias->dimensions[0];
+  params.bias_type = bias->type;
+
+  if (params.input_type == mojom::TENSOR_QUANT8_ASYMM ||
+      params.input_type == mojom::TENSOR_QUANT8_ASYMM_SIGNED) {
+    float input_scale = input->scale;
+    float dst_scale = output->scale;
+    if (params.filter_type == mojom::TENSOR_QUANT8_SYMM_PER_CHANNEL) {
+      // get weight scales
+      mojom::OperandSymmPerChannelQuantParamPtr& filter_param =
+          model_info_->operandSymmPerChannelQuantParams[filter_idx];
+      int32_t channel_dim = filter_param->channel_dim;
+      if ((channel_dim == 0 && !params.depthwise) ||
+          (channel_dim == 3 && params.depthwise)) {
+        params.scales_params.output_mask = 1 << 1;
```

Comment:
I think the code that sets up the `DNNLInt8ScalesParams` should move to compilation_delegate_dnnl.cc.

Reply:
I agree with you and move the related code to compilation_delegate_dnnl.cc.